lib/auth/github.go

```diff
@@ -622,6 +622,11 @@ func (a *Server) validateGithubAuthCallback(ctx context.Context, diagCtx *SSODia
 		return nil, trace.Wrap(err)
 	}
 
+	userState, err := a.GetUserOrLoginState(ctx, user.GetName())
```

Comment:
Do you intend to make this same change for SAML and OIDC in the enterprise repo? It looks like `validateOIDCAuthCallback()` and `validateSAMLResponse()` also create the web session with values from a `User` resource and not a `UserState`.

Reply:
I do, that's actually the reason I made this `GetUserOrLoginState` public here.